src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -308,6 +310,20 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
                 result.RestoreMetadata.CentralPackageVersionOverrideDisabled = isCentralPackageVersionOverrideDisabled;
                 result.RestoreMetadata.CentralPackageFloatingVersionsEnabled = isCentralPackageFloatingVersionsEnabled;
                 result.RestoreMetadata.CentralPackageTransitivePinningEnabled = isCentralPackageTransitivePinningEnabled;
+
+                string isSdk = specItem.GetProperty("UsingMicrosoftNETSdk");
+
+                if (!string.IsNullOrEmpty(isSdk))
+                {
+                    if (string.Equals(isSdk, "true", StringComparison.OrdinalIgnoreCase))
+                    {
+                        result.UsingMicrosoftNETSdk = true;
+                    }
+                    else if (string.Equals(isSdk, "false", StringComparison.OrdinalIgnoreCase))
+                    {
+                        result.UsingMicrosoftNETSdk = false;
+                    }
```

Comment:
I didn't check the syntax, so this might not compile as is

```suggestion
                    if (bool.TryParse(isSdk, out bool value))
                    {
                        result.UsingMicrosoftNETSdk = value;
                    }
```